### PR TITLE
Woo/add product tags

### LINKED ISSUE #1613: Woo: Product Tags - add endpoint to add new tags
As part of woocommerce/woocommerce-android#2190, we need to add the option to add a new product tag to a site.

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #1613 by adding support to add a new tag.

#### Changes
- Added new action and action classes to `WCProductStore` and `ProductRestClient` to add a new product tag for a site.
- Added unit tests.
- Added a new button to the example app to add a new product tag.

#### Notes
~- **This PR is in draft till this [PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1617) can be reviewed and merged.**~

#### To Test
- Run `MockedStack_WCProductsTest` and `ReleaseStack_WCProductTest`.
- In the example app, click on `Woo` -> `Products` - > `Select Site` - > `Add Product Tag`.
- Enter a unique tag name and verify that the tag is added successfully to the test site.
- Now click on `Add Product Tag` again and enter the same tag name from the previous step. Notice that an error message is displayed: `TERM_EXISTS`. This is because the API throws an error when we try to update the same tag name.

### CODE REVIEW

File: example/src/androidTest/java/org/wordpress/android/fluxc/mocked/MockedStack_WCProductsTest.kt [hunk 1]
```diff
@@ -772,6 +774,31 @@ class MockedStack_WCProductsTest : MockedStack_Base() {
         assertNotNull(payload.error)
     }
 
+    @Test
+    fun testAddProductTagSuccess() {
```
Comment (AmandaRiu) [inline example/src/androidTest/java/org/wordpress/android/fluxc/mocked/MockedStack_WCProductsTest.kt:778]: np: I think it'd be good to add a `testAddProductTagFailed()` to verify the `isError` is properly being populated.

Comment (AmandaRiu) [inline example/src/androidTest/java/org/wordpress/android/fluxc/mocked/MockedStack_WCProductsTest.kt:778]: np: I recommend adding a test for duplicate tag handling as well.

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_WCProductTest.kt [hunk 2]
```diff
@@ -550,6 +553,32 @@ class ReleaseStack_WCProductTest : ReleaseStack_WCBase() {
...
+        ProductSqlUtils.deleteProductTagsForSite(sSite)
+        assertEquals(0, ProductSqlUtils.getProductTagsForSite(sSite.id).size)
+
+        nextEvent = TestEvent.ADDED_PRODUCT_TAG
+        mCountDownLatch = CountDownLatch(1)
+
+        val productTagModel = WCProductTagModel().apply {
+            // duplicate tag names fail in the API level so added a random number next to the "Test"
+            name = "Test" + Random.nextInt(0, 10000)
+        }
```
Comment (AmandaRiu) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_WCProductTest.kt:569]: Adding a random number may still fail periodically. Maybe appending the current date in milliseconds would be more full-proof.

**Comment by anitaa1990** [comment]
Hey @AmandaRiu! Thanks so much for the review! I found out yesterday that we need batch update tags for edit products M3 and that involves calling another endpoint (`/wc/v3/products/tags/batch`). So we no longer need to implement the endpoint from this PR 🤦‍♀️ 

I'm closing this for now and will open another one for that PR. Apologies for wasting your time!
